apps/studio/src/lib/utility/appdb/SavedConnection.ts

```diff
@@ -0,0 +1,118 @@
+import { ConnectionType, IConnection, SshMode } from "@/common/interfaces/IConnection";
+import { AzureAuthOptions, BigQueryOptions, CassandraOptions, LibSQLOptions, RedshiftOptions } from "@/common/appdb/models/saved_connection";
+import { BaseUtilityAppDbEntity } from "./BaseUtilityAppDbEntity";
+import { ConnectionString } from "connection-string";
+import rawLog from 'electron-log';
+import { baseFind, baseFindOne } from "./Util";
+
+const log = rawLog.scope('SavedConnection');
+
+export class SavedConnection extends BaseUtilityAppDbEntity implements IConnection {
```

Comment:
So the idea here is right, but we already have an abstraction for this because we needed one for Workspaces. It's done at the store level, rather than the model level.

I think you need to extend those interfaces, rather than introducing a second competing abstraction.

See for example `CloudClient`, `CloudConnectionModule`, and `DataModuleBase`